### PR TITLE
Update client to 2.2.5. Take 2

### PR DESCRIPTION
**Pull request by marcthe12**
I have Update binary to version 2.2.5

I have also clean up the unnecessary workarounds such as:

- Remove xdotool upstream has fixed the issue

- Libpng and libglew dependency has been dropped as the client does not depend on it

I also improve the audio workaround as it was giving shellcheck errors.

TODO: figure out how to get CAP_SETFCAP set on the binary which allows us to drop the current audio workaround

### CODE REVIEW

File: scripts/runescape.sh [hunk 1]
```diff
@@ -2,21 +2,9 @@
...
+export PULSE_LATENCY_MSEC
 
-if [ -z "${SHOW_LAUNCHER}" ]; then
-    # The game won't be up instantly. So wait until we see 2 visible RuneScape windows.
-    # While we're waiting, grab the first window ID. This will be the launcher window.
-    until [ $(xdotool search --all --onlyvisible --classname "RuneScape" | wc -l) -eq 2 ]; do
-        [ "$launcherId" ] || launcherId=$(xdotool search --all --onlyvisible --classname "RuneScape")
-        sleep 2
-    done
+ exec /app/extra/runescape-launcher "$@"
```
Comment (AsciiWolf) [inline scripts/runescape.sh]: Small typo: Trailing space before `exec`.

**Comment by marcthe12** [comment]
tested it and works

**Comment by AsciiWolf** [comment]
I have tested this as well. Seems to work fine!

**Comment by marcthe12** [comment]
Btw do you if flatpak sandbox allows CAP_SETFCAP. Can't find documentation on this.

**Comment by MrCarroll** [comment]
Flatpak internally uses LD_PRELOAD to tell LD, the system linker, where to find the dependencies that the game is looking for. When LD detects that a binary has a CAP_* set, it will disable the LD_PRELOAD functionality to prevent security issues that could come from it. This means that granting the capabilities that the binary needs to fix the audio without using the workaround will render the Flatpak unusable entirely and cannot be worked around. I have been trying to use rtkit-daemon as a replacement, however I do not think this is a valid route to fixing the audio scheduling either.

Unfortunately, Flatpak and even Snap are incompatible with the proper audio fix.

**Comment by marcthe12** [comment]
Well we can't do anything. The best we can do is to talk to flatpak Devs or get jagex us rtkit on this.

**Pull request merged**